connectivity/netsocket/include/netsocket/nsapi_types.h

```diff
@@ -406,6 +407,22 @@ typedef struct nsapi_stagger_req {
     uint16_t stagger_rand;  /* [OUT] Randomized stagger value in seconds */
 } nsapi_stagger_req_t;
 
+/** nsapi_msghdr
+ */
+typedef struct nsapi_msghdr {
+    nsapi_size_t len;    /* Data byte count, including header */
+    int          level;  /* Originating protocol */
+    int          type;   /* Protocol-specific type */
+} nsapi_msghdr_t;
+
+/** nsapi_pktinfo structure
+ */
+typedef struct nsapi_pktinfo {
+    nsapi_msghdr_t hdr;         /* Header identifying the message control structure */
+    nsapi_addr_t   ipi_addr;    /* Address associated with the packet */
+    int            ipi_ifindex; /* Interface associated with the packet */
```

Comment:
This will return the internal interface index. I believe we should return/accept a pointer to a `NetworkInterface`. Some translation must be done in the NetworkStack implementation to make that possible.

Reply:
I believe this chunk of API is currently C...  That might be rather intractible.

For starters, we do need some way of mapping IDs to/from `NetworkInterface`s. I think that is currently missing - we've hit this before. A comparator is the multicast group membership sockopt.

Reply:
Done